frontend/pages/sign-up.tsx

```diff
@@ -1,197 +1,231 @@
+                  error={formik.touched.username && Boolean(formik.errors.username)}
+                  helperText={formik.touched.username && formik.errors.username}
+                />
+                <TextField required fullWidth margin="normal"  variant="outlined"
+                  id="emailField"
+                  name="email"
+                  label="Email"
+                  InputLabelProps={{shrink: true}}
+                  value={formik.values.email}
+                  onChange={formik.handleChange}
+                  error = {formik.touched.email && Boolean(formik.errors.email)}
+                  helperText={formik.touched.email && formik.errors.email}
+                />
+                <TextField required fullWidth  margin="normal" variant="outlined"
+                  id="password1Field"
+                  name="password1"
+                  type="password"
+                  label="Password"
+                  InputLabelProps={{shrink: true,}}
+                  value={formik.values.password1}
+                  onChange={formik.handleChange}
+                  error={formik.touched.password1 && Boolean(formik.errors.password1)}
+                  helperText={formik.touched.password1 && formik.errors.password1}
+                />
+                <TextField required fullWidth margin="normal" variant="outlined"
+                  id="password2Field"
+                  name="password2"
+                  type="password"
+                  label="Confirm Password"
+                  InputLabelProps={{shrink: true}}
+                  value={formik.values.password2}
+                  onChange={formik.handleChange}
+                  error={formik.touched.password2 && Boolean(formik.errors.password2)}
+                  helperText={formik.touched.password2 && formik.errors.password2}
+                />
+
+                <Button type="submit" variant="contained" fullWidth color="primary" size="large">
+                  Sign Up
+                </Button>
+
+              </form>
+
+            <Typography variant="body2" align="center">
+            <div>
+              { errorState && (
+                <Alert severity="error">Username or Email is already in use</Alert>
+              )}
+            </div>
+          </Typography>
+
           </Container>
         </div>
       </PageContainer>
     </React.Fragment>
   );
 }
 
+const REGISTER_USER_MUTATION = gql
+  `
+    mutation register(
+      $academicTitle: String!
+      $firstName: String!,
+      $lastName: String!,
+      $username: String!,
+      $email: String!,
+      $password1: String!,
+      $password2: String!
+      ) {
+        register(
+          academicTitle: $academicTitle,
+          firstName: $firstName,
+          lastName: $lastName,
+          username: $username,
+          email: $email,
+          password1: $password1,
+          password2: $password2
+        )
+          {
+         success
+         errors
+      }
+    }
+  `;
+
```

Comment:
Is there a certain reason or design pattern behind movin the mutations to the bottom of the files?
If not so, I would suggest either leaving them on the top of the files or moving all the mutations to the bottom. I don't care which way we do it, I would just like some consistency :)

Reply:
Ah, I just did this so that I wouldn't have to scroll too far down when working on the form. I'll revert this

Reply:
done

Reply:
great